### DISCUSSION TITLE
Force Connection Port back to Client?

### DISCUSSION THREAD

**Discussion by austin-InDro** [bluenviron/mediamtx#1248] [Questions]
Hello,

I'm curious if it is possible to force a connection Port back to the client? I currently have your system running on a computer behind a NAT. I have opened up port 8554 to connect to the system. However, when the client attempts to connect back, a new random port is created and the connection is blocked.

Do you happen to know a solution to this?

[RTSP Stream]----> [RTSP Simple Server Computer] ----> [NAT with Port 8554 open and connected to 8554 on Computer] ---> [Client}

**Comment by aler9** [comment]
Hello, this issue happens when you use the UDP transport protocol, that can't be used when there's a NAT between clients and server.  You have to switch to the TCP transport protocol. You can read more about transport protocols here:

https://github.com/aler9/rtsp-simple-server#rtsp-protocol

**Reply by austin-InDro** [reply]
Great thanks!

[a comment by an automated account is not shown]
